test/unit/fact_cleaner_test.rb

```diff
@@ -48,6 +48,49 @@ class FactCleanerTest < ActiveSupport::TestCase
     assert_include remaining_of(root_6_fact), root_6_fact
   end
 
+  test "it cleans excluded facts" do
+    cleaner.stubs(:excluded_facts).returns(['ignored*', '*bad'])
+    cleaner.stubs(:delete_orphaned_facts).returns(0)
+    good_fact_name = FactoryBot.create(:fact_name, :name => 'good_fact')
+    ignored_fact_name = FactoryBot.create(:fact_name, :name => 'ignored01')
+    bad_fact_name = FactoryBot.create(:fact_name, :name => 'empty_bad')
+    FactoryBot.create(:fact_value, :fact_name => good_fact_name)
+    FactoryBot.create(:fact_value, :fact_name => ignored_fact_name)
```

Comment:
I still do not see testing of "macaddress_em1" kind of fact. Frankly, I was hoping for real-world interface names rather than these artificial ones :-)